src/main/java/dev/warriorrr/simplertp/PlayerListener.java

```
@@ -22,6 +23,7 @@ public void onSpawnLocationEvent(PlayerSpawnLocationEvent event) {
             return;
 
         final Location location = plugin.generator().getAndRemove();
+        if (location == null) return;
```

The location returned by the generated should never be null, I'd rather we get a stacktrace than silently returning and having players spawn at 0, 0 again

---

The method doesn't return null. if safeLocations is empty it returns the world spawn, idk if thats at 0 0
Removed this return statement